ios/RNLinksdk.m

```diff
@@ -48,6 +50,42 @@ @interface RNLinksdk ()
 
 #pragma mark -
 
+@interface PLKSuccessMetadataTypoFix : NSObject
+
+@end
+
+@implementation PLKSuccessMetadataTypoFix
+
++ (void)load {
+    Class targetClass = NSClassFromString(@"PLKSuccessMetadata");
+    SEL typoSel = NSSelectorFromString(@"insitution");
+    SEL correctSel = NSSelectorFromString(@"institution");
+
+    BOOL respondsToCorrectSel = class_respondsToSelector(targetClass, correctSel);
+    BOOL respondsToTypoSel = class_respondsToSelector(targetClass, typoSel);
+
+    if (respondsToCorrectSel && respondsToTypoSel) {
+        return;
+    }
+
```

Comment:
Suggestion to clarify the next bits by using a helper variable `missingSel` and `existingSel`:
```
  if (!respondsToCorrectSel && !respondsToTypoSel) {
        // throw the github issues error
    }
    
SEL missingSel = (respondsToTypoSel) ? correctSel : typoSel
SEL existingSel = (respondsToTypoSel) ? typoSel : correctSel 

```

Helps with the DRY

Reply:
@pladaub I really like @amytang0's suggestion as it makes the change much more readable and should allow to shorten the code to:

```
+ (void)load {
    Class targetClass = NSClassFromString(@"PLKSuccessMetadata");
    SEL typoSel = NSSelectorFromString(@"insitution");
    SEL correctSel = NSSelectorFromString(@"institution");

    BOOL respondsToTypoSel = class_respondsToSelector(targetClass, typoSel);
    BOOL respondsToCorrectSel = class_respondsToSelector(targetClass, correctSel);

    if (!(respondsToCorrectSel || respondsToTypoSel)) {
        NSString *githubIssueURLString = @"https://github.com/plaid/react-native-plaid-link-sdk/issues/new?assignees=&labels=&template=bug_report.md&title=";
        NSAssert(false, @"PLKSuccessMetadata does not respond to correctly spelled %@ or legacy, typo %@. This is a bug in either react-native-plaid-link-sdk, or LinkKit. Please file an issue at: %@", NSStringFromSelector(correctSel), NSStringFromSelector(typoSel), githubIssueURLString);
    }

    SEL missingSel = (respondsToTypoSel) ? correctSel : typoSel;
    SEL existingSel = (respondsToTypoSel) ? typoSel : correctSel;
    Method method = class_getInstanceMethod(targetClass, existingSel);
    const char* types = method_getTypeEncoding(method);
    IMP implementation = class_getMethodImplementation(targetClass, existingSel);
    class_addMethod(targetClass, missingSel, implementation, types);
}
```

Reply:
Yes, very much like this. We still need to check that swizzling isn't necessary, but this is a big readability improvement.